packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -121,6 +162,46 @@ void main() {
     expect(box.size.height - oldHeight, greaterThanOrEqualTo(100.0)); // 100 + some margin
   });
 
+  testWidgets('ExpansionPanelList does not merge header when canTapOnHeader is false', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: ExpansionPanelListSemanticsTest(),
+      ),
+    );
+    final SemanticsHandle handle = tester.ensureSemantics();
+    await tester.pump();
+    // Make sure it is clickable.
+    await tester.tap(find.text('head1'));
+    await tester.pump();
+
+    final ExpansionPanelListSemanticsTestState state =
+      tester.state(find.byType(ExpansionPanelListSemanticsTest));
+    expect(state.headerTapped, true);
+
+    final Finder expandedIcon = find.descendant(
+      of: find.ancestor(
+        of: find.byKey(headerKey),
+        matching: find.byType(Row),
+      ),
+      matching: find.byType(ExpandIcon),
+    );
+    expect(tester.getSemantics(expandedIcon), matchesSemantics(
+      isButton: true,
+      hasEnabledState: true,
+      isEnabled: true,
+      hasTapAction: true,
+    ));
+    final Finder headerWidget = find.descendant(
+      of: find.byKey(headerKey),
+      matching: find.byType(RichText),
+    );
+    expect(tester.getSemantics(headerWidget), matchesSemantics(
+      label: 'head1',
+      hasTapAction: true,
```

Comment:
Why do they both have a tap action? I thought the point of this is that only the button is tappable, not the header?

Reply:
I simulate the situation where user have a custom gesture detector in the header, so it has both tap at the head and icon, they are two separate semantics nodes

Reply:
See comment above. That should probably be cleared up in a comment.

Also, it would be good to have a test that shows that the header doesn't have a click handler if canTapOnHeader is set to false.